js/compiler/igor-controller.js

```diff
@@ -13,16 +12,24 @@ import { spawn } from '../GMConstructor.js';
 /** @type {Job[]} */
 const jobs = [];
 
+const fs = require("node:fs/promises");
+
 /**
  * Run a new job on a given project.
  * @param {GMLProject} project
  * @param {RuntimeInfo} runtime
+ * @param {UserInfo?} user
  * @param {IgorSettings} settings
  * @returns {Promise<Result<Job>>}
  */
-export async function job_run(project, runtime, settings) {
+export async function job_run(project, runtime, user, settings) {
+
+    // for some reason if we don't clear the cache directory, changes won't apply in yyc
+    if (settings.runner === "YYC") {
+        await fs.rm(project.dir + "/cache/", {recursive: true, force: true});
```

Comment:
I'm not super comfortable with using what's basically an `rm -rf` here, I'd rather leave that to Igor to manage which it already does via a `Clean` task. If this is needed I'd argue for invoking a `Clean` and on completion invoking the actual compile job. I'd want to do more experimenting on this too since the cache is a big part of YYC speed and initial compilation on YYC is horrendously slow.